services/storage/src/simcore_service_storage/modules/celery/worker.py

```diff
@@ -0,0 +1,32 @@
+import logging
+from collections.abc import Callable
+
+from celery import Celery
+from celery.contrib.abortable import AbortableTask
+from models_library.progress_bar import ProgressReport
+
+from .models import TaskID
+
+_logger = logging.getLogger(__name__)
+
+
+class CeleryTaskQueueWorker:
+    def __init__(self, celery_app: Celery) -> None:
+        self.celery_app = celery_app
+
+    def register_task(self, fn: Callable, task_name: str | None = None) -> None:
```

Comment:
do we want unnamed tasks?

Reply:
Not really unamed, but rather will take the name of the function: `name = task_name or fn.__name__`